### PR TITLE
Snackbar opacity animation shouldn't trigger on reverse

### LINKED ISSUE #617: Snackbar opacity animation shouldn't trigger on reverse
We should give it a reverse curve that stays at 1.0 until t=0.0 then drops to 0.0.

### PR DESCRIPTION
**Pull request by abarth**
Fixes #617

### CODE REVIEW

**Comment by abarth** [comment]
@HansMuller @Hixie

File: packages/flutter/lib/src/animation/curves.dart [hunk 1]
```diff
@@ -95,6 +95,25 @@ class Interval extends Curve {
   }
 }
 
+/// A curve that is 0.0 until step, then jumps to 1.0.
+class Step extends Curve {
+  const Step(this.step);
```
Comment (HansMuller) [inline packages/flutter/lib/src/animation/curves.dart]: Maybe this parameter should be called threshold?

**Comment by HansMuller** [comment]
LGTM

File: packages/flutter/lib/src/animation/curves.dart (hunk 1 above)
Comment (abarth) [inline packages/flutter/lib/src/animation/curves.dart]: Ah, good idea.

**Pull request merged**